document new ingestion/indexing ootb dashboard APM (#13858)
DataDog/documentation: content/en/tracing/trace_ingestion/usage_metrics.md

---
title: Usage Metrics
kind: documentation
aliases:
    - /tracing/trace_retention_and_ingestion/usage_metrics/
    - /tracing/trace_retention/usage_metrics/
description: "Learn how to monitor your APM usage."
further_reading:
- link: "/tracing/trace_ingestion/"
  tag: "Documentation"
  text: "Trace Ingestion"
- link: "/tracing/trace_retention/"
  tag: "Documentation"
  text: "Trace Retention"
---

## Overview

> To control APM Ingested and indexed volumes, use the two dedicated in-app configuration pages:

Reviewer: ```suggestion
The following in-app configuration pages allow you to set ingested and indexed volumes for APM:
```

> - The **[Ingestion Control Page][1]** to control the ingested span volume.

Reviewer: ```suggestion
- See the **[Ingestion Control Page][1]** to control ingested span volume.
```

> - The **[Retention Filters page][2]** to control the number of indexed spans.

Reviewer: ```suggestion
- See the **[Retention Filters page][2]** to control the number of indexed spans.
```

> Both pages are powered by **usage metrics**. These metrics are available in your account:

Reviewer: ```suggestion
Both pages are powered by **usage metrics**. 

The following metrics are available in your account:

```

 - `datadog.estimated_usage.apm.ingested_bytes` (billed dimension)
 - `datadog.estimated_usage.apm.ingested_spans`
 - `datadog.estimated_usage.apm.ingested_traces`
 - `datadog.estimated_usage.apm.indexed_spans` (billed dimension)

> 
> Leverage these metrics in **dashboards** and **monitors** to visualise and control your usage or alert when it goes over a certain limit. Two out-of-the box dashboards are built based on these metrics. These dashboards help you to monitor your APM usage as well as your ingested and indexed span volumes.

Reviewer: ```suggestion

Leverage these metrics in dashboards and monitors with visualizations, controls on your usage, and alerts when metrics exceed a certain limit. Two out-of-the box dashboards are built with these metrics. These dashboards help 
 monitor your APM usage as well as your ingested and indexed span volumes.
```

> Datadog APM plans come with included indexed and ingested spans. For more information, read the [pricing documentation][3], or view some [example pricing scenarios][4].

Reviewer: ```suggestion
Datadog APM plans come with indexed and ingested spans included. For more information, see the [Pricing page][3] or some [pricing example scenarios][4].
```

## Usage Metrics

### Ingested spans volume

> 3 usage metrics are tied to the ingested spans usage:

Reviewer: ```suggestion
The following metrics are associated with ingested spans usage:

```

 - `datadog.estimated_usage.apm.ingested_bytes`
 - `datadog.estimated_usage.apm.ingested_spans`
 - `datadog.estimated_usage.apm.ingested_traces`

> Use `datadog.estimated_usage.apm.ingested_bytes` to control your usage as ingestion is metered as a **volume** and not as a number of spans or traces. This metric is tagged by `env` and `service` to allow you to spot which environments and services are contributing to the ingestion volume.

Reviewer: ```suggestion
Control the usage with `datadog.estimated_usage.apm.ingested_bytes`. Ingestion is metered as a volume, not as a number of spans or traces. This metric is tagged by `env` and `service` so you can spot which environments and services are contributing to the ingestion volume.
```

> This metric is also tagged by `ingestion_reason`: this dimension reflects which [ingestion mechanisms][5] are responsible for sending the spans to Datadog. These mechanisms are nested in the tracing libraries of in the Datadog Agent. This dimension is introduced in details in the [Ingestion Reason][6] dashboard.

Reviewer: ```suggestion
This metric is also tagged by `ingestion_reason`, reflecting which [ingestion mechanisms][5] are responsible for sending spans to Datadog. These mechanisms are nested in the tracing libraries of the Datadog Agent. For more information about this dimension, see the [Ingestion Reason dashboard][6].
```

> The `datadog.estimated_usage.apm.ingested_traces` metric measures the number of requests sampled per second. This only counts traces sampled by the [head-based sampling][7]. This metric is also tagged by `env` and `service`. Use this metric to spot which services are starting the most traces.

Reviewer: ```suggestion
The `datadog.estimated_usage.apm.ingested_traces` metric measures the number of requests sampled per second, and only counts traces sampled by [head-based sampling][7]. This metric is also tagged by `env` and `service` so you can spot which services are starting the most traces.
```

### Indexed spans

Use the `datadog.estimated_usage.apm.indexed_spans` metric to control the number of spans indexed by [tag-based retention filters][2].

> This metric is tagged by `env` and `service` to allow you to spot which environments and services are contributing to the indexing usage.

Reviewer: ```suggestion
This metric is tagged by `env` and `service` so you can spot which environments and services are contributing to the indexing usage.
```

> ## APM Traces Estimated Usage dashboard

Reviewer: ```suggestion
## Estimated usage dashboard
```

> The [APM Traces Usage dashboard][8] has several groups of widgets to see at a glance high level KPIs, and more detailed information about the usage.

Reviewer: ```suggestion
The [APM Traces Usage dashboard][8] contains several widget groups displaying high-level KPIs and additional  usage information.
```

{{< img src="tracing/trace_indexing_and_ingestion/usage_metrics/dashboard_apm_usage.png" style="width:100%;" alt="APM Estimated Usage Dashboard" >}}

> Thanks to the dashboard, get visibility over :
> - Your **global usage**
> - Your **infrastructure** with APM enabled (hosts, Fargate, AWS Lambda)
> - Your **ingestion** volumes, with breakdowns by `service`, `env`, and `ingestion_reason`
> - Your **indexing** volumes, with breakdowns by `service`, `env`

Reviewer: ```suggestion
In this dashboard, you can find information about:

- Global usage metrics
- Infrastructure with APM enabled, including hosts, Fargate, and AWS Lambda
- Ingestion volumes separated by `service`, `env`, and `ingestion_reason`
- Indexing volumes separated by `service` and `env`
```
